perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/PerunBlImpl.java

```diff
@@ -153,6 +171,34 @@ public PerunSession getPerunSession(PerunPrincipal principal, PerunClient client
 		return perunSession;
 	}
 
+	/**
+	 * Iteratively searches through additional identifiers trying to find UES with the same identifier.
+	 * Returns first found UES or null when no matching UES found.
+	 *
+	 * @param principal PerunPrincipal which contains additionalIdentifiers
+	 * @param internalSession PerunSession to retrieve UserExtSource
+	 * @return UserExtSource found using additionalIdentifiers or null
+	 * @throws InternalErrorException When attribute is not defined in Perun
+	 */
+	public UserExtSource getUserExtSourceFromMultipleIdentifiers(PerunPrincipal principal, PerunSession internalSession) throws InternalErrorException {
```

Comment:
It is standard for us to have session always as the first parameter if possible. And the name is normally just "session" or "sess".

Reply:
Understood and fixed.